libcodechecker/libclient/client.py

```diff
@@ -29,7 +32,20 @@ def check_api_version(client):
     """
 
     version = client.getAPIVersion()
-    supp_major_version = SUPPORTED_VERSION.split('.')[0]
+    supp_major_version = SUPPORTED_RUN_API_VERSION.split('.')[0]
+    api_major_version = version.split('.')[0]
+
+    # There is NO compatibility between major versions.
+    return supp_major_version == api_major_version
+
+
+def check_product_api_version(client):
+    """
+    Check if server product API is supported by the client.
+    """
+
+    version = client.getAPIVersion()
```

Comment:
Yes. If a client can make sure through every other means that the product exists (and later on, the user has rights to access), then it need not access the product-api at all.

But... then again, is it likely that we will have clients not our own? :grin:

I'll eliminate these and put the version into the shared file. But the code that checks the version here must stay, because this function and the other can be called entirely separately. (One client is the one that returns the same version from endpoint `Products`, the other from `CodeCheckerService`.) Two distinct control flow, even though the value right now is the same, and even if the value will be made _actually_ the same constant all the time.